src/Renderer/LayeredMaterial.js

```diff
@@ -77,10 +62,8 @@ const LayeredMaterial = function LayeredMaterial(options) {
     THREE.RawShaderMaterial.call(this);
 
     const nbSamplers = getMaxColorSamplerUnitsCount();
-    this.vertexShader = TileVS;
 
-    this.fragmentShaderHeader = `${PrecisionQualifier}\nconst int   TEX_UNITS   = ${nbSamplers.toString()};\n`;
-    this.fragmentShaderHeader += pitUV;
+    this.fragmentShaderHeader = `#define TEX_UNITS ${nbSamplers.toString()}\n`;
```

Comment:
Could you use `this.defines.TEX_UNITS = nbSamplers;` instead of creating 
`this.fragmentShaderHeader` ?